conda-store-server/conda_store_server/__init__.py

```diff
@@ -86,14 +107,28 @@ def get_build_key(cls, build: "Build") -> str:  # noqa: F821
         return cls._fmt.get(build.build_key_version)(build)
 
     @classmethod
-    def parse_build_key(cls, build_key: str) -> int:
+    def parse_build_key(
+        cls, conda_store: "CondaStore", build_key: str  # noqa: F821
+    ) -> int:
         """Returns build id from build key"""
+        # This import is here to avoid cyclic imports
+        from conda_store_server import orm
+
         parts = build_key.split("-")
         # Note: cannot rely on the number of dashes to differentiate between
         # versions because name can contain dashes. Instead, this relies on the
         # hash size to infer the format. The name is the last field, so indexing
         # to find the id is okay.
         if build_key[cls._version2_hash_size] == "-":  # v2
             return int(parts[2])  # build_id
+        elif "-" not in build_key:  # v3
+            with conda_store.get_db() as db:
+                builds = db.query(orm.Build).all()
+                # Note: this uses a separate for-loop instead of using filter
+                # because the latter doesn't allow for passing lambdas
+                for build in builds:
+                    if cls._version3_fmt(build) == build_key:
+                        return build.id
```

Comment:
Are there any performance threats here? Should we return _all_ builds ever or maybe do it in chunks?